packages/discovery-provider/src/tasks/entity_manager/entities/playlist.py

```diff
@@ -146,20 +146,60 @@ def update_playlist_tracks(params: ManageEntityParameters, playlist_record: Play
         if playlist_track.track_id not in updated_track_ids:
             playlist_track.is_removed = True
             playlist_track.updated_at = params.block_datetime
-            track = (
+            track_record = (
                 session.query(Track)
                 .filter(Track.track_id == playlist_track.track_id)
                 .first()
             )
-            if track:
-                track.updated_at = params.block_datetime
-                track.playlists_containing_track = [
-                    collection_id
-                    for collection_id in (track.playlists_containing_track or [])
-                    if collection_id != playlist["playlist_id"]
+            track = helpers.model_to_dictionary(track_record)
+            if track and track_record:
+                track_record.updated_at = params.block_datetime
+                # remove from playlists_containing_track
+                track_record.playlists_containing_track = [
+                    playlist_id
+                    for playlist_id in track["playlists_containing_track"]
+                    if playlist_id != playlist["playlist_id"]
                 ]
 
+                # add to or overwrite existing entry in playlists_previously_containing_track
+                if "playlists" not in track["playlists_previously_containing_track"]:
+                    continue
+                track_record.playlists_previously_containing_track = {
+                    "playlists": [
```

Comment:
not to jump into a religious debate about [lambda/filter vs comprehension](https://stackoverflow.com/questions/3013449/list-comprehension-vs-lambda-filter), but using filter also would work if you were looking for an alternative

Reply:
fair, we actually wrote some lambdas before but then realized that they're barely used in our codebase so decided to follow existing patterns